index.d.cts

```diff
@@ -0,0 +1,54 @@
+declare const _exports: PrettierSync;
+export = _exports;
+export type Prettier = typeof import("prettier");
+export type PrettierSync = PrettierSyncFunctions & PrettierStaticProperties;
+export type PrettierSyncFunctions = {
+  formatWithCursor: PrettierSyncFunction<"formatWithCursor">;
+  format: PrettierSyncFunction<"format">;
+  check: PrettierSyncFunction<"check">;
+  resolveConfig: PrettierSyncFunction<"resolveConfig">;
+  resolveConfigFile: PrettierSyncFunction<"resolveConfigFile">;
+  clearConfigCache: PrettierSyncFunction<"clearConfigCache">;
+  getFileInfo: PrettierSyncFunction<"getFileInfo">;
+  getSupportInfo: PrettierSyncFunction<"getSupportInfo">;
+};
+export type PrettierStaticProperties = {
+  version: string;
+  util: typeof import("prettier").util;
+  doc: typeof import("prettier").doc;
+};
+export type PrettierSyncFunction<
+  FunctionName extends
+    | "formatWithCursor"
+    | "format"
+    | "check"
+    | "resolveConfig"
+    | "resolveConfigFile"
+    | "clearConfigCache"
+    | "getFileInfo"
+    | "getSupportInfo",
+> = (
+  ...args: Parameters<Prettier[FunctionName]>
+) => Awaited<ReturnType<Prettier[FunctionName]>>;
+/**
+ * @param {object} options
+ * @param {string | URL} options.prettierEntry - Path or URL to prettier entry.
+ * @returns {PrettierSync}
+ */
+export function createSynchronizedPrettier({
+  prettierEntry,
+}: {
+  prettierEntry: string | URL;
+}): PrettierSync;
+declare const PRETTIER_ASYNC_FUNCTIONS: readonly [
+  "formatWithCursor",
+  "format",
+  "check",
+  "resolveConfig",
+  "resolveConfigFile",
+  "clearConfigCache",
+  "getFileInfo",
+  "getSupportInfo",
+];
+declare const PRETTIER_STATIC_PROPERTIES: readonly ["version", "util", "doc"];
```

Comment:
I'm not familiar with types, but why are these needed?

Reply:
They're not publicly exported, but I've added `/** @type {const} */` to these two:

```mjs
PRETTIER_ASYNC_FUNCTIONS
PRETTIER_STATIC_PROPERTIES
```

When a function is called or property is accessed that doesn't exist (or renamed) it'll validate against:

```mjs
FunctionName in typeof PRETTIER_ASYNC_FUNCTIONS[number]
PropertyName in typeof PRETTIER_STATIC_PROPERTIES[number]
```

Otherwise we'd have to duplicate them in quite a few places:

```ts
FunctionName extends "formatWithCursor" | "format" | "check" | "resolveConfig" | "resolveConfigFile" | "clearConfigCache" | "getFileInfo" | "getSupportInfo"
Property extends "version" | "util" | "doc"
```

These consts are used to populate the functions/properties available in editors:

<img width="638" alt="Prettier available functions and properties" src="https://github.com/prettier/prettier-synchronized/assets/415517/fd8b9cae-411a-427f-8eb7-e24d137ef0fa">

Reply:
Appreciate they're not needed externally so tried adding `@internal`

But when setting **tsconfig.json** `"stripInternal": true` it didn't work due to:

* https://github.com/microsoft/TypeScript/issues/46407
* https://github.com/microsoft/TypeScript/issues/46515

One of the downsides to using the compiler with JSDoc 😔